command/operator_debug.go

```diff
@@ -1714,3 +1740,54 @@ func isRedirectError(err error) bool {
 	const redirectErr string = `invalid character '<' looking for beginning of value`
 	return strings.Contains(err.Error(), redirectErr)
 }
+
+// getNomadVersion fetches the version of Nomad running on a given server/client node ID
+func (c *OperatorDebugCommand) getNomadVersion(serverID string, nodeID string) string {
+	if serverID == "" && nodeID == "" {
+		return ""
+	}
+
+	version := ""
+	if serverID != "" {
+		for _, server := range c.members.Members {
```

Comment:
`c.members` is being concurrently accessed by multiple pprof goroutines. I _think_ it's only ever written once and then read multiple times but this is brittle. But also we're calling this function on every periodic pprof iteration, of which there can be many. Instead, we can do this work before we start spawning pprof goroutines (skipping the goroutines entirely for those agents that we shouldn't pprof).

Reply:
Passing this as an argument to the function would let the compiler enforce the happens-before dependency, but we don't need to block on that.